astropy/modeling/mappings.py

```diff
@@ -34,8 +38,8 @@ def mapping(self):
         return self._mapping
 
     def evaluate(self, *args):
-        if len(args) < self.n_inputs:
-            raise TypeError('{0} expects at most {1} inputs; got {2}'.format(
+        if len(args) != self.n_inputs:
+            raise TypeError('{0} expects {1} inputs; got {2}'.format(
```

Comment:
Sure this will work in `evaluate`. But then you have to pass `n_inputs=n` in order to be able to chain them correctly. And this makes it strict.

 I think it can become tricky and hard to debug so being explicit is not a bad thing in this case.